protocol/x/clob/keeper/deleveraging.go

```diff
@@ -241,6 +242,7 @@ func (k Keeper) OffsetSubaccountPerpetualPosition(
 				*offsettingSubaccount.Id,
 				perpetualId,
 				deltaQuantums,
+				false,
```

Comment:
nit: I would pull the branching logic out of `ProcessDeleveraging` and just pass in `fillPriceDeltaQuoteQuantums` to this function. 

personally not a fan of using booleans as function arguments in general since it kind of ruins code readability. wdyt?

Reply:
hmm, I'd argue it is a little cleaner from a code perspective to push this branching down into `ProcessDeleveraging`. If we have to pass in the fill price to the function then we need to calculate this anywhere we call `ProcessDeleveraging`. Right now that is in both PrepareCheckState and DeliverTx (for persisting the deleveraging match). This will mean the code for calculating fill price or bankruptcy price will be written twice in the code base instead of just once in the `ProcessDeleveraging` function. Also makes those top level functions that call `ProcessDeleveraging` more clear to read as more of the deleveraging logic is abstracted away

Reply:
curious how strong your opinion on this is

Reply:
i guess deltaQuantums is currently passed in to ProcessDeleveraging though. Could see this going either way.

Reply:
Slight preference for pulling this out. Having the calculation not dependent on a boolean flag is also more future proof in case we need to settle at a price other than bankruptcy/oracle price. 

Your argument about having the logic twice makes sense. Maybe we can define a helper function to minimize duplicated code, something like `GetDeleveragingSettlementPrice(bool)`?

We might not even need that boolean since if TNC < 0 we will use bankruptcy, and oracle otherwise. but we don't need to worry about this now